nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,408 @@
+                    .explanation("Error while parsing the schema: " + e.getMessage())
+                    .build());
+            return problems;
+        }
+        return super.customValidate(validationContext);
+    }
+
+    @OnScheduled
+    public void setPreference(final ProcessContext context) {
+        this.preference.set(new CsvPreference.Builder(context.getProperty(QUOTE_CHARACTER).getValue().charAt(0),
+                context.getProperty(DELIMITER_CHARACTER).getValue().charAt(0),
+                context.getProperty(END_OF_LINE_CHARACTER).getValue()).build());
+    }
+
+    /**
+     * Method used to parse the string supplied by the user. The string is converted
+     * to a list of cell processors used to validate the CSV data.
+     * @param schema Schema to parse
+     */
+    private void parseSchema(String schema) {
+        this.processors.set(new ArrayList<CellProcessor>());
+        String remaining = schema;
+        while(remaining.length() > 0) {
+            remaining = setProcessor(remaining);
+        }
+    }
+
+    private String setProcessor(String remaining) {
+        StringBuffer buffer = new StringBuffer();
+        int i = 0;
+        int opening = 0;
+        int closing = 0;
+        while(buffer.length() != remaining.length()) {
+            char c = remaining.charAt(i);
+            i++;
+
+            if(opening == 0 && c == ',') {
+                if(i == 1) {
+                    continue;
+                }
+                break;
+            }
+
+            buffer.append(c);
+
+            if(c == '(') {
+                opening++;
+            } else if(c == ')') {
+                closing++;
+            }
+
+            if(opening > 0 && opening == closing) {
+                break;
+            }
+        }
+
+        final String procString = buffer.toString().trim();
+        opening = procString.indexOf('(');
+        String method = procString;
+        String argument = null;
+        if(opening != -1) {
+            argument = method.substring(opening + 1, method.length() - 1);
+            method = method.substring(0, opening);
+        }
+
+        this.processors.get().add(getProcessor(method.toLowerCase(), argument));
+
+        return remaining.substring(i);
+    }
+
+    private CellProcessor getProcessor(String method, String argument) {
+        switch (method) {
+
+        case "optional":
+            int opening = argument.indexOf('(');
+            String subMethod = argument;
+            String subArgument = null;
+            if(opening != -1) {
+                subArgument = subMethod.substring(opening + 1, subMethod.length() - 1);
+                subMethod = subMethod.substring(0, opening);
+            }
+            return new Optional(getProcessor(subMethod.toLowerCase(), subArgument));
+
+        case "parsedate":
+            return new ParseDate(argument.substring(1, argument.length() - 1));
+
+        case "parsedouble":
+            return new ParseDouble();
```

Comment:
Since this method is being used to validate, all of these no-param methods should validate that there are no arguments being passed to them.